openvdb_cmd/vdb_tool/include/Parser.h

```diff
@@ -0,0 +1,1037 @@
+        add("pow3","cube of value, e.g. {2:pow3} -> {8}",
+            [&](){this->a([](auto& x){return math::Pow3(x);});});
+        add("pow","power of vale, e.g. {2:3:pow} -> {8}",
+            [&](){this->ab([](auto& a, auto& b){return math::Pow(a, b);});});
+        add("min","minimum of two values, e.g. {1:2:min} -> {1}",
+            [&](){this->ab([](auto& a, auto& b){return std::min(a, b);});});
+        add("max","minimum of two values, e.g. {1:2:max} -> {2}",
+            [&](){this->ab([](auto& a, auto& b){return std::max(a, b);});});
+        add("neg","negative of value, e.g. {1:neg} -> {-1}",
+            [&](){this->a([](auto& x){return -x;});});
+        add("sign","sign of value, e.g. {-2:neg} -> {-1}",
+            [&](){this->a([](auto& x){return (x > 0) - (x < 0);});});
+        add("sin","sine of value, e.g. {$pi:sin} -> {0.0}",
+            [&](){this->set(std::sin(str2float(mCallStack.top())));});
+        add("cos","cosine of value, e.g. {$pi:cos} -> {-1.0}",
+            [&](){this->set(std::cos(str2float(mCallStack.top())));});
+        add("tan","tangent of value, e.g. {$pi:tan} -> {0.0}",
+            [&](){this->set(std::tan(str2float(mCallStack.top())));});
+        add("asin","inverse sine of value, e.g. {1:asin} -> {1.570796}",
+            [&](){this->set(std::asin(str2float(mCallStack.top())));});
+        add("acos","inverse cosine of value, e.g. {1:acos} -> {0.0}",
+            [&](){this->set(std::acos(str2float(mCallStack.top())));});
+        add("atan","inverse tangent of value, e.g. {1:atan} -> {0.785398}",
+            [&](){this->set(std::atan(str2float(mCallStack.top())));});
+        add("r2d","radian to degrees, e.g. {$pi:r2d} -> {180.0}",
+            [&](){this->set(180.0f*str2float(mCallStack.top())/math::pi<float>());});
+        add("d2r","degrees to radian, e.g. {180:d2r} -> {3.141593}",
+            [&](){this->set(math::pi<float>()*str2float(mCallStack.top())/180.0f);});
+        add("inv","inverse of value, e.g. {5:inv} -> {0.2}",
+            [&](){this->set(1.0f/str2float(mCallStack.top()));});
+        add("exp","exponential of value, e.g. {1:exp} -> {2.718282}",
+            [&](){this->set(std::exp(str2float(mCallStack.top())));});
+        add("ln","natural log of value, e.g. {1:ln} -> {0.0}",
+            [&](){this->set(std::log(str2float(mCallStack.top())));});
+        add("log","10 base log of value, e.g. {1:log} -> {0.0}",
+            [&](){this->set(std::log10(str2float(mCallStack.top())));});
+        add("sqrt","squareroot of value, e.g. {2:sqrt} -> {1.414214}",
+            [&](){this->set(std::sqrt(str2float(mCallStack.top())));});
+        add("to_int","convert value to integer, e.g. {1.2:to_int} -> {1}",
+            [&](){this->set(int(str2float(mCallStack.top())));});
+        add("to_float","convert value to floating point, e.g. {1:to_float} -> {1.0}",
+            [&](){this->set(str2float(mCallStack.top()));});
```

Comment:
Why do you need all these functions? As far as I understand it you can only use these with loop variables? I can't see any examples using math operations though maybe I've missed them

Reply:
you can define your own variables using "@", e.g. "{1:@x}" is semantically equivalent to "x=1". See "# String evaluation:" in README.md for an (extreme) examples where it's used to solve a quadratic equation. So in short you can define your own local variables that can then be used to set any of the options for the various actions. Very useful indeed

Reply:
let's sat you want to use some procedure/function to define the value of an option, like the voxel-size of a grid, or radius or center of level set sphere, or even the frame number of a file you want to load. There are tons of example - think of all the expressions you can write in Houdini to define parameters in the GUI. This is the same but applied to option of the command-line actions :)

Reply:
Why not just use the systems python or something... This is also exactly what AX can do and doesn't require new syntax (which looks pretty difficult to understand)